sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -328,6 +339,12 @@ def enable_slep006():
   to the splitter
 - method_args: a dict of dicts, defining extra arguments needed to be passed to
   methods, such as passing `classes` to `partial_fit`.
+- requests_set_together: a dict that defines which set_{method}_requests need
+  to be set together with the key; used in case a router routes to different
+  methods from the sub-estimator from withing the same meta-estimator's method.
+  For instance, {"fit": ["score"]} would signal that
+  `estimator.set_fit_request` premises `estimator.set_score_request` to be set
+  as well.
```

Comment:
But if we include the one-to-one mapping, in this case fit to fit, then we would need to a check if `method_mapping` exists very early of the test and have a big devide of how to deal with the two different cases.

I had thought about this before, and came to the conclusion that it would make the test more complicated. After all, it's better to treat this further down in the test. This is why I designed it this way. Would you mind approving it as it is, @adrinjalali?